lib/univalue.cpp

```diff
@@ -233,12 +233,15 @@ const char *uvTypeName(UniValue::VType t)
     return NULL;
 }
 
-const UniValue& find_value(const UniValue& obj, const std::string& name)
+const UniValue& find_value(const UniValue& obj, const std::string& name, const bool throw_if_not_present)
 {
-    for (unsigned int i = 0; i < obj.keys.size(); i++)
+    for (unsigned int i = 0; i < obj.keys.size(); i++) {
         if (obj.keys[i] == name)
             return obj.values.at(i);
+    }
 
+    if (throw_if_not_present) {
+        throw int(-1);
```

Comment:
I think in general we want to throw exceptions derived from `std::exception`. Univalue seems to use `runtime_exception` for other cases, so this may be sufficient.